### ISSUE TITLE
Undefined Variables: Section 3 Rule 3.5.3.2.3

### ISSUE THREAD

**Issue by estenrye** [darkwizard242/cis_ubuntu_2004#7]
# Role Version

2.0.0

# Undefined Default Variables

```yaml
    ubuntu_2004_cis_section3_rule_3_5_3_2_1_params_iptableschain_input_name: INPUT
    ubuntu_2004_cis_section3_rule_3_5_3_2_1_params_iptableschain_input_policy: DROP
    ubuntu_2004_cis_section3_rule_3_5_3_2_1_params_iptableschain_output_name: OUTPUT
    ubuntu_2004_cis_section3_rule_3_5_3_2_1_params_iptableschain_output_policy: DROP
    ubuntu_2004_cis_section3_rule_3_5_3_2_1_params_iptableschain_forward_name: FORWARD
    ubuntu_2004_cis_section3_rule_3_5_3_2_1_params_iptableschain_forward_policy: DROP
```

# Error Encountered

```yaml
    vmware.vsphere-iso.vmware: TASK [darkwizard242.cis_ubuntu_2004 : 3.5.3.2.3 | Ensure iptables default deny firewall policy (Automated) | set input chain to drop] ***
    vmware.vsphere-iso.vmware: fatal: [default]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'ubuntu_2004_cis_section3_rule_3_5_3_2_1_params_iptableschain_input_name' is undefined

The error appears to be in '/root/.ansible/roles/darkwizard242.cis_ubuntu_2004/tasks/section_03.yml': line 952, column 7, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:
    - name: "3.5.3.2.3 | Ensure iptables default deny firewall policy (Automated) | set input chain to drop"
```

# Playbook Example

```yaml
- hosts: default
  become: true
  vars:
    ubuntu_2004_cis_auditd_maxlogfile: 32
    ubuntu_2004_cis_time_synchronization: chrony
    ubuntu_2004_cis_firewall: iptables
    ubuntu_2004_cis_require_ssh_ciphers: aes256-ctr,aes192-ctr,aes128-ctr
    ubuntu_2004_cis_require_ssh_macs: hmac-sha2-256,hmac-sha2-512
    ubuntu_2004_cis_require_ssh_kexalgorithms: ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group14-sha256
    ubuntu_2004_cis_require_ssh_allowusers: ubuntu
    ubuntu_2004_cis_require_ssh_allowgroups: ubuntu
    ubuntu_2004_cis_require_ssh_denyusers: bogus
    ubuntu_2004_cis_require_ssh_denygroups: bogus
  tasks:
    - name: Set vagrant usernames in allowed SSH
      ansible.builtin.set_fact:
        ubuntu_2004_cis_require_ssh_allowusers: ubuntu  vagrant
        ubuntu_2004_cis_require_ssh_allowgroups: ubuntu  vagrant
      when:
        - packer_builder_type == "virtualbox-iso"

    - name: execute cis_ubuntu_2004 role
      ansible.builtin.include_role:
        name: darkwizard242.cis_ubuntu_2004
```

# Line Where the Error Occurs

https://github.com/darkwizard242/cis_ubuntu_2004/blob/812611571fb7ba2da8d3ec7412d17467dc143280/tasks/section_03.yml#L952-L956

https://github.com/darkwizard242/cis_ubuntu_2004/blob/812611571fb7ba2da8d3ec7412d17467dc143280/tasks/section_03.yml#L958-L962

https://github.com/darkwizard242/cis_ubuntu_2004/blob/812611571fb7ba2da8d3ec7412d17467dc143280/tasks/section_03.yml#L964-L968

# Workaround

```yaml
- hosts: default
  become: true
  vars:
    ubuntu_2004_cis_auditd_maxlogfile: 32
    ubuntu_2004_cis_time_synchronization: chrony
    ubuntu_2004_cis_firewall: iptables
    ubuntu_2004_cis_require_ssh_ciphers: aes256-ctr,aes192-ctr,aes128-ctr
    ubuntu_2004_cis_require_ssh_macs: hmac-sha2-256,hmac-sha2-512
    ubuntu_2004_cis_require_ssh_kexalgorithms: ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group14-sha256
    ubuntu_2004_cis_require_ssh_allowusers: ubuntu
    ubuntu_2004_cis_require_ssh_allowgroups: ubuntu
    ubuntu_2004_cis_require_ssh_denyusers: bogus
    ubuntu_2004_cis_require_ssh_denygroups: bogus
    ubuntu_2004_cis_section3_rule_3_5_3_2_1_params_iptableschain_input_name: INPUT
    ubuntu_2004_cis_section3_rule_3_5_3_2_1_params_iptableschain_input_policy: DROP
    ubuntu_2004_cis_section3_rule_3_5_3_2_1_params_iptableschain_output_name: OUTPUT
    ubuntu_2004_cis_section3_rule_3_5_3_2_1_params_iptableschain_output_policy: DROP
    ubuntu_2004_cis_section3_rule_3_5_3_2_1_params_iptableschain_forward_name: FORWARD
    ubuntu_2004_cis_section3_rule_3_5_3_2_1_params_iptableschain_forward_policy: DROP
  tasks:
    - name: Set vagrant usernames in allowed SSH
      ansible.builtin.set_fact:
        ubuntu_2004_cis_require_ssh_allowusers: ubuntu vagrant
        ubuntu_2004_cis_require_ssh_allowgroups: ubuntu vagrant
      when:
        - packer_builder_type == "virtualbox-iso"

    - name: execute cis_ubuntu_2004 role
      ansible.builtin.include_role:
        name: darkwizard242.cis_ubuntu_2004
```

**Comment by darkwizard242** [comment]
@estenrye - Good catch 👍🏼 Thanks for identifying that.

Will rectify that, release a fixed version and share.

The following six vars will be changed:

**FROM**:
```yaml
ubuntu_2004_cis_section3_rule_3_5_3_2_1_params_iptableschain_input_name
ubuntu_2004_cis_section3_rule_3_5_3_2_1_params_iptableschain_input_policy
ubuntu_2004_cis_section3_rule_3_5_3_2_1_params_iptableschain_output_name
ubuntu_2004_cis_section3_rule_3_5_3_2_1_params_iptableschain_output_policy
ubuntu_2004_cis_section3_rule_3_5_3_2_1_params_iptableschain_forward_name
ubuntu_2004_cis_section3_rule_3_5_3_2_1_params_iptableschain_forward_policy
```

**TO**:
```yaml
ubuntu_2004_cis_section3_rule_3_5_3_2_3_params_iptableschain_input_name
ubuntu_2004_cis_section3_rule_3_5_3_2_3_params_iptableschain_input_policy
ubuntu_2004_cis_section3_rule_3_5_3_2_3_params_iptableschain_output_name
ubuntu_2004_cis_section3_rule_3_5_3_2_3_params_iptableschain_output_policy
ubuntu_2004_cis_section3_rule_3_5_3_2_3_params_iptableschain_forward_name
ubuntu_2004_cis_section3_rule_3_5_3_2_3_params_iptableschain_forward_policy
```

**Comment by darkwizard242** [comment]
@estenrye - Fixed, merged and released [2.0.1](https://github.com/darkwizard242/cis_ubuntu_2004/releases/tag/2.0.1). Available now on Ansible Galaxy.

Thanks again for identifying that. I have been using `ufw` instead of `iptables` on most of my systems and haven't had a thorough validation done around `iptables` yet. So, your findings really are of great help to this project.

Feel free to close the issue upon confirmation from your side for this issue.

**Comment by estenrye** [comment]
Looks like that is fixed.
